### ISSUE TITLE
support blpop in older version of redis <v6.0.0

### ISSUE THREAD

**Issue by lambdaq** [vectordotdev/vector#22061]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Problem

https://github.com/vectordotdev/vector/blob/029a2ffc33c66fe1d8b755ad309b695317b1f542/src/sources/redis/list.rs#L75

`conn.brpop(key, 0.0)`

`conn.blpop(key, 0.0)`

https://redis.io/docs/latest/commands/brpop/

> Starting with Redis version 6.0.0: timeout is interpreted as a double instead of an integer.

Unfortunately we have several old redis instances running and can't use vector for this exact reason

My config

```
[sources.in]
type = "redis"
data_type = "list"
list.method = "brpop"
key = "sth"
url = "redis://xxxx"
```

> vector::internal_events::redis: Failed to read message. error=An error was signalled by the server - ResponseError: timeout is not an integer or out of range error_code=ERR error_type="reader_failed" stage="sending" internal_log_rate_limit=true

### Version

latest

**Comment by jszwedko** [comment]
Thanks @lambdaq ! I think this would need to be a change in the redis library. I opened https://github.com/redis-rs/redis-rs/issues/1453 if you want to follow along there.

**Comment by lambdaq** [comment]
> Redis' [version lifetime support](https://redis.io/docs/latest/operate/rs/installing-upgrading/product-lifecycle/), which means that 6 is already out of support and 6.2 is soon out of support.

Ahh damn. Guess I have to write my own redis blpop code without vector.

**Comment by jszwedko** [comment]
Indeed, it does look like the library we are using only aims to maintain compatibility with supported versions of Redis. This is typically our aim with Vector integrations too: maintaining support for supported versions of data sources and destinations to keep our maintenance burden manageable. Given that, I think we'll have to close this issue. Apologies that Vector can't support this use-case!
